qemu/tests/virtio_serial_hotplug_max_chardevs.py

```diff
@@ -0,0 +1,108 @@
+import logging
+
+from virttest import error_context
+from virttest import utils_misc
+from virttest import utils_test
+from virttest.qemu_devices.utils import DeviceHotplugError
+
+from qemu.tests import driver_in_use
+from qemu.tests.virtio_console import add_chardev
+from qemu.tests.virtio_console import add_virtio_ports_to_vm
+from qemu.tests.virtio_serial_file_transfer import transfer_data
+from qemu.tests.virtio_serial_hotplug_port_pci import get_buses_and_serial_devices
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    Hot-plug max chardevs on one virtio-serial-pci
+
+    1. Boot a guest without any device
+    2. Hotplug virtio-serial-pci
+    3. Hotplug 31 chardevs
+    4. Hotadd 30 virtserialports attached every one chardev
+    5. Transfer data between guest and host via all ports
+    6. Repeat hotplug existed one chardev
+    7. Repeat hotplug existed one virtserialport
+    8. Hot-unplug virtserialport
+    9. Hot-unplug chardev
+
+    :param test: kvm test object
+    :param params: Dictionary with the test parameters
+    :param env: Dictionary with test environment
+    """
+    def run_serial_data_transfer():
+        """
+        Transfer data between two ports.
+        """
+
+        for serial_port in serials:
+            port_params = params.object_params(serial_port)
+            if not port_params['serial_type'].startswith('virtserial'):
+                continue
+            logging.info("transfer data with port %s" % serial_port)
+            params['file_transfer_serial_port'] = serial_port
+            transfer_data(params, vm, sender='both')
+
+    def run_bg_test():
+        """
+        Set the operation of transferring data as background
+        :return: return the background case thread if it's successful;
+                 else raise error
+        """
+        stress_thread = utils_misc.InterruptedThread(run_serial_data_transfer)
+        stress_thread.start()
+        if not utils_misc.wait_for(lambda: driver_in_use.check_bg_running(
+                vm, params), check_bg_timeout, 0, 1):
+            test.fail("Backgroud test is not alive!")
+        return stress_thread
+
+    vm = env.get_vm(params["main_vm"])
```

Comment:
Could you add the step content?
error_context.context()

Reply:
updated already, please help to check again, thanks.